### PR TITLE
Function to test if a given category is a valid input for Freyd category

### PR DESCRIPTION
**Pull request by HereAround**
The following methods are also used in our derivations of methods for Freyd categories:
- IsCongruentForMorphisms
- IsEqualForMorphisms
- ZeroObjectFunctorial
- DirectSumFunctorial

However, currently these are not demanded by an underlying category (i.e. in the function/tester introduced in this PR). We should discuss this.

### CODE REVIEW

File: FreydCategoriesForCAP/gap/FreydCategory.gi [hunk 1]
```diff
@@ -19,64 +19,47 @@ InstallMethod( FreydCategory,
...
-    if ForAll( [ "WeakKernelEmbedding", "WeakKernelLift" ], f -> CanCompute( underlying_category, f ) ) then
-        
-        SetIsAbelianCategory( freyd_category, true );
-        
-    fi;
-    
-    AddObjectRepresentation( freyd_category, IsFreydCategoryObject );
-    
-    AddMorphismRepresentation( freyd_category, IsFreydCategoryMorphism );
+    if IsValidInputForFreydCategory( underlying_category ) then
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategory.gi]: Could you change the logic to:
```
if not IsValidInputForFreydCategory( underlying_category ) then
    return false;
fi;
```
and then adjust the indenting accordingly?
In this way, the code is easier for me to read.

File: FreydCategoriesForCAP/gap/FreydCategory.gi [hunk 2]
```diff
@@ -1055,3 +1038,69 @@ InstallMethod( Display,
...
+                      "UniversalMorphismIntoDirectSum",
+                      "InjectionOfCofactorOfDirectSum",
+                      "UniversalMorphismFromDirectSum",
+                      "Lift"
+                      ];
+    installed_ops := ListInstalledOperationsOfCategory( category );
+
+    for i in required_ops do
+
+      if not i in installed_ops then
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategory.gi]: Could you use the command `CanCompute` in this line please?

**Comment by HereAround** [comment]
The above commit provides the suggested changes.

File: FreydCategoriesForCAP/gap/FreydCategory.gi (hunk 1 above)
Comment (HereAround) [reply FreydCategoriesForCAP/gap/FreydCategory.gi]: Yes, the commit below provides this change.

File: FreydCategoriesForCAP/gap/FreydCategory.gi (hunk 2 above)
Comment (HereAround) [reply FreydCategoriesForCAP/gap/FreydCategory.gi]: Yes, the commit below provides this change.

File: FreydCategoriesForCAP/gap/FreydCategory.gi [hunk 3]
```diff
@@ -19,57 +19,36 @@ InstallMethod( FreydCategory,
...
-    if not CanCompute( underlying_category, "PreCompose" ) then
-        
-        Error( "The given category should be able to compute PreCompose" );
-        
-    fi;
-    
+
     freyd_category := CreateCapCategory( Concatenation( "Freyd( ", Name( underlying_category ), " )" ) );
-    
+
```
Comment (sebastianpos) [inline FreydCategoriesForCAP/gap/FreydCategory.gi]: Just a nitpick: your editor seems to have deleted the whitespace in the empty lines.

If you don't want to "repair" this, I can also do this later.

Comment (HereAround) [reply FreydCategoriesForCAP/gap/FreydCategory.gi]: Strange, this was not even displayed by git on my local system. Indeed took a little effort to remove all these white space alternations. If I missed any (which I hope I did not), then please consider merging this PR and make the changes afterwards.

**Pull request merged**